src/function.rs

```diff
@@ -205,10 +206,42 @@ impl<'lua> Function<'lua> {
             Ok(Function(lua.pop_ref()))
         }
     }
+
+    /// Returns source and line number information about the function.
+    ///
+    /// Corresponds to the `>S` what mask for `lua_getinfo` when applied
+    /// to the function.
+    pub fn source(&self) -> FunctionSource {
+        let lua = self.0.lua;
+        unsafe {
+            let mut ar: ffi::lua_Debug = std::mem::zeroed();
+            lua.push_ref(&self.0);
+            mlua_assert!(
+                ffi::lua_getinfo(lua.state, cstr!(">S"), &mut ar as *mut ffi::lua_Debug) != 0,
+                "lua_getinfo failed with `>S`"
+            );
+            FunctionSource {
+                source: String::from_utf8_lossy(CStr::from_ptr(ar.source as *const i8).to_bytes()).into_owned(),
```

Comment:
`String::from_utf8_lossy` could lose data for non-utf8 encoded names.

Consider the two tests:
```
    lua.load("function function_6() end")
    .set_name(&[103, 102, 101])?
    .exec()?;
    let function_6 = globals.get::<_, Function>("function_6")?;
    let function_6_source = function_6.source();
    assert_eq!(function_6_source.source.as_bytes(), &[103, 102, 101]);
```
and
```
    lua.load("function function_6() end")
    .set_name(&[203, 202, 201])?
    .exec()?;
    let function_6 = globals.get::<_, Function>("function_6")?;
    let function_6_source = function_6.source();
    assert_eq!(function_6_source.source.as_bytes(), &[203, 202, 201]);
```

The 2nd one will fail. 
Would be better to define `source` and `short_src` as `Vec<u8>` ? This would allow to users to choose between `String::from_utf8_lossy()` or `str::from_utf8`.

Also, `c_char` could be [`u8`](https://doc.rust-lang.org/std/os/raw/type.c_char.html) on some platforms, and `CStr::from_ptr(ar.source as *const i8)` could be rewritten to `CStr::from_ptr(ar.source)` since `CStr::from_ptr` accepts `c_char` and `ar.source` itself is `c_char`.